targets/csharp/source/source/Uunit/tests/PlayFabApiTest.cs

```diff
@@ -537,10 +480,169 @@ private void GetObjects2Continued(PlayFabResult<GetObjectsResponse> result, UUni
+            var loginTask2 = PlayFabClientAPI.LoginWithCustomIDAsync(loginRequest2, null, testTitleData.extraHeaders).Result;
+
+            testContext.NotNull(loginTask.Result, "Login Result is null for loginRequest");
+            testContext.NotNull(loginTask2.Result, "Login Result is null for loginRequest2");
+            testContext.IsNull(loginTask.Error, "Login error occured for loginRequest: " + loginTask.Error?.ErrorMessage ?? string.Empty);
+            testContext.IsNull(loginTask2.Error, "Login error occured for loginRequest2: " + loginTask2.Error?.ErrorMessage ?? string.Empty);
+            testContext.NotNull(loginTask.Result.AuthenticationContext, "AuthenticationContext is not set for loginRequest");
+            testContext.NotNull(loginTask2.Result.AuthenticationContext, "AuthenticationContext is not set for loginRequest2");
+
+            if (string.Equals(loginTask.Result.AuthenticationContext.ClientSessionTicket, loginTask2.Result.AuthenticationContext.ClientSessionTicket))
+            {
+                testContext.Fail("MultipleApiCall Failed AuthenticationContexts are same");
+            }
+            var getPlayerProfileRequest = new GetPlayerProfileRequest()
+            {
+                AuthenticationContext = loginTask.Result.AuthenticationContext,
+                PlayFabId = PlayFabId
+            };
+            var getPlayerProfileRequest2 = new GetPlayerProfileRequest()
+            {
+                AuthenticationContext = loginTask2.Result.AuthenticationContext,
+                PlayFabId = PlayFabId
+            };
+
+            var getPlayerProfileTask = PlayFabClientAPI.GetPlayerProfileAsync(getPlayerProfileRequest, null, testTitleData.extraHeaders).Result;
+            var getPlayerProfileTask2 = PlayFabClientAPI.GetPlayerProfileAsync(getPlayerProfileRequest2, null, testTitleData.extraHeaders).Result;
+
+            testContext.NotNull(getPlayerProfileTask.Result, "GetPlayerProfile Failed for getPlayerProfileRequest");
+            testContext.NotNull(getPlayerProfileTask2.Result, "GetPlayerProfile Failed for getPlayerProfileRequest2");
+            testContext.IsNull(getPlayerProfileTask.Error, "GetPlayerProfile error occured for getPlayerProfileRequest: " + getPlayerProfileTask.Error?.ErrorMessage ?? string.Empty);
+            testContext.IsNull(getPlayerProfileTask2.Error, "GetPlayerProfile error occured for getPlayerProfileRequest2: " + getPlayerProfileTask2.Error?.ErrorMessage ?? string.Empty);
+
+            testContext.EndTest(UUnitFinishState.PASSED, null);
+        }
+
+        /// <summary>
+        /// CLIENT API
+        /// Tests async api calls with multiple users
+        /// </summary>
+        [UUnitTest]
+        public void AsyncApiCallWithMultipleUser(UUnitTestContext testContext)
+        {
+            var loginRequest = new LoginWithCustomIDRequest
+            {
+                TitleId = PlayFabSettings.TitleId,
+                CustomId = PlayFabSettings.BuildIdentifier,
```

Comment:
This test seems very weird.
1. This test should be using three different players, and you're using the same one three times.

Reply:
I did not changed this method yet because if I should test with 3 different player i need 3 different titleId. Am I wrong? I will modify this method after clarifying it in my mind

Reply:
Melih, i think you can use different CustomIds to distinguish them

---
In reply to: [257152426](https://github.com/PlayFab/SDKGenerator/pull/404#discussion_r257152426) [](ancestors = 257152426)

Reply:
Oh thanks! I will do this ASAP

Reply:
I commit it now. Does it meet the needs?